tools/benchmark/cmd/storage-put.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2015 Nippon Telegraph and Telephone Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"crypto/rand"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/coreos/etcd/Godeps/_workspace/src/github.com/spf13/cobra"
+)
+
+// storagePutCmd represents a storage put performance benchmarking tool
+var storagePutCmd = &cobra.Command{
+	Use:   "put",
+	Short: "Benchmark put performance of storage",
+
+	Run: storagePutFunc,
+}
+
+var (
+	nrKeys         int
+	storageKeySize int
+	valueSize      int
+	txn            bool
+)
+
+func init() {
+	storageCmd.AddCommand(storagePutCmd)
+
+	storagePutCmd.Flags().IntVar(&nrKeys, "nr-keys", 100, "a number of keys to put")
+	storagePutCmd.Flags().IntVar(&storageKeySize, "storage-key-size", 64, "a size of key (Byte)")
+	storagePutCmd.Flags().IntVar(&valueSize, "value-size", 64, "a size of value (Byte)")
+	storagePutCmd.Flags().BoolVar(&txn, "txn", false, "put a key in transaction or not")
+}
+
+func createBytesSlice(bytesN, sliceN int) [][]byte {
+	rs := make([][]byte, sliceN)
+	for i := range rs {
+		rs[i] = make([]byte, bytesN)
+		if _, err := rand.Read(rs[i]); err != nil {
+			panic(err)
+		}
+	}
+	return rs
+}
+
+func storagePutFunc(cmd *cobra.Command, args []string) {
+	keys := createBytesSlice(storageKeySize, nrKeys)
```

Comment:
actually we should only bench seq put. etcd only use backend to do sequential put instead of random put.